src/PrestaShopBundle/Controller/Admin/Improve/Design/MailThemeController.php

```diff
@@ -0,0 +1,179 @@
+
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\Form\FormHandlerInterface;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Form\Admin\Improve\Design\MailTheme\GenerateMailsType;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use PrestaShopBundle\Service\MailTemplate\GenerateMailTemplatesService;
+use Symfony\Component\Form\Form;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class MailThemeController manages mail theme generation, you can define the shop
+ * mail theme, and regenerate mail in a specific language.
+ *
+ * Accessible via "Design > Mail Theme"
+ */
+class MailThemeController extends FrameworkBundleAdminController
+{
+    /**
+     * Show mail theme settings and generation page.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))", message="Access denied.")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+        $defaultMailTheme = $this->configuration->get('PS_MAIL_THEME');
+        $generateThemeMailsForm = $this->createForm(GenerateMailsType::class, ['theme' => $defaultMailTheme]);
+
+        return $this->render('@PrestaShop/Admin/Improve/Design/MailTheme/index.html.twig', [
+            'layoutHeaderToolbarBtn' => [],
+            'layoutTitle' => $this->trans('Mail Theme', 'Admin.Navigation.Menu'),
+            'requireAddonsSearch' => true,
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($legacyController),
+            'mailThemeConfigurationForm' => $this->getMailThemeFormHandler()->getForm()->createView(),
+            'generateMailsForm' => $generateThemeMailsForm->createView(),
+        ]);
+    }
+
+    /**
+     * Manage generation form post and generate mails.
+     *
+     * @AdminSecurity("is_granted(['create', 'update'], request.get('_legacy_controller'))", message="Access denied.")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function generateMailsAction(Request $request)
+    {
+        $generateThemeMailsForm = $this->createForm(GenerateMailsType::class);
+        $generateThemeMailsForm->handleRequest($request);
+
+        if ($generateThemeMailsForm->isSubmitted()) {
+            $data = $generateThemeMailsForm->getData();
+
+            try {
+                /** @var GenerateMailTemplatesService $generator */
+                $generator = $this->get('prestashop.service.generate_mail_templates');
+                //Overwrite theme folder if selected
+                if (!empty($data['theme'])) {
+                    $themeFolder = $this->getParameter('kernel.project_dir') . '/themes/' . $data['theme'];
+                    $generator
```

Comment:
you will laugh (thinking it's me who are saying this): shouldn't be a call from the commandBus here instead?

Reply:
I don't see the link with commandBus or CQRS, as we are not dealing with database changes

Reply:
isn't a writing operation? CQRS is not related to MySQL, it's related to view and write operations :)

Reply:
Oh ok, as CQRS stand for `Command Query Responsibility Segregation` I guess I was influenced by the `Query` part and thought of it as Database purpose mainly.
I didn't imagine this could be used to file generation as well.
That said does it justify to block this feature? Because this won't be done in time for 176

Reply:
Done